src/vm/jithelpers.cpp

```diff
@@ -4427,69 +4427,18 @@ NOINLINE static void JIT_MonEnter_Helper(Object* obj, BYTE* pbLockTaken, LPVOID
 }
 
 /*********************************************************************/
-NOINLINE static void JIT_MonContention_Helper(Object* obj, BYTE* pbLockTaken, LPVOID __me)
```

Comment:
I am trying to rationalize the value of this MonContention helper.     This is uniformly called as a second last chance' before bailing and calling the 'slow' MonEnter_Helper path.    Why do we not simply skip this and call the slow helper and be done with it?   At this point we don't care about CPU as we need to kill some time anyway.

Reply:
Not sure I follow, as that's kind of what I did, I removed the MonContention helper. The main purpose of it before the changes was to repeat the entire spin loop 10 more times before waiting. I suspect it was necessary mainly to avoid lock convoys, as spinning more helps to avoid them with the previous design. After the changes, there is a first-level fast path check for the unlocked case (one try for the lock), followed by a spin loop of a default of 30 iterations, then the slow path that waits.

Reply:
Sorry, my mistake ignore my comment.   I was looking at the wrong branch on my machine and confused myself about that fact that this code is being removed.